seatunnel-server/seatunnel-app/src/main/java/org/apache/seatunnel/app/service/impl/UserServiceImpl.java

```diff
@@ -160,10 +165,38 @@ public UserSimpleInfoRes login(UserLoginReq req) {
                         .userId(user.getId())
                         .build();
         userDaoImpl.insertLoginLog(logDto);
-
         return translate;
     }
 
+    private User ldapAuthentication(UserLoginReq req) {
+        String username = req.getUsername();
+        String password = req.getPassword();
+        Authentication authenticationRequest =
+                new UsernamePasswordAuthenticationToken(username, password);
+        try {
+            configuredAuthenticationProvider.authenticate(authenticationRequest);
+        } catch (AuthenticationException ex) {
+            throw new SeatunnelException(USERNAME_PASSWORD_NO_MATCHED);
+        }
+
+        if (userDaoImpl.getByName(username) == null) {
+            // 2. add a new user.
+            final UpdateUserDto dto =
+                    UpdateUserDto.builder().id(null).username(username).password("").build();
+            userDaoImpl.add(dto);
```

Comment:
After the first login, an LDAP user is added to the database. From the second login onwards, the same user can log in as a DB_USER by just not providing Auth-Type header.
There should be a way to differentiate between user types.